pkg/endpoint: Consistently use logrus fields

raybejjani opened the pull request:
This switches a bunch of log callsites to use logrus.WIthFields. These changes are not supposed to break anything and follow the same reasoning from https://github.com/cilium/cilium/pull/1801 The goal of these changes is to make debugging via logs easier by making our usage more consistent. Ideally, field names should be used the same way throughout the code and mean the same thing.

The most important things to review are whether the field names make sense in the context they're used, and whether we should introduce new ones to pkg/logfields/logfields.go. I'm also happy to incorporate better messages into this PR :)
I might have been too aggressive by making a logger method bound to endpoint. I'm not sure if the fields reported are always appropriate, but they seemed to be. In particular, I'm not sure if Iteration is relevant, and if I'm reporting it correctly in any case.

raybejjani committed:
pkg/endpoint: Consistently use logrus fields

aanm reviewed (approved):
Just make sure the `log()` method is always called with the  endpoint's RLock locked

aanm commented on pkg/endpoint/log.go:
```diff
@@ -0,0 +1,37 @@
+	})
+}
+
+func (e *Endpoint) getIteration() uint64 {
```
@tgraf should be `e.Consumable.Iteration` or `e.policyRevision`?

tgraf replied:
The iteration is unused now, it predated policy revision. We can remove the iteration field and use the policy revision in the logs.

raybejjani committed:
Fixup endpoint.log to be in lock path (except Endpoint.SetIdentity)

raybejjani committed:
pkg/endpoint: Unexport Endpoint critical path functions

These functions were exported, despite not being used outside the
package. They need to be locked to acquire the Endpoint logger, or where
marked *Locked to begin with.

raybejjani committed:
daemon: Daemon.SyncState locks Endpoint to syncLabels

Daemon.syncLabels access parts of the Endpoint structure. These are
protected by Endpoint.Mutex and we should hold the lock to read them.
